.github/workflows/initi.yaml

```diff
@@ -0,0 +1,141 @@
+name: CI
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  ghr_push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare
+        id: prep
+        uses: ./.github/actions/docker-image-tag
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.REP_PAT }}
+
+      -  name: Build and push
+         uses: docker/build-push-action@v2
+         with:
+           context: .
+           file: ./Dockerfile
+#          push: ${{ github.event_name != 'pull_request' }}
+           push: true
+           tags: ${{ steps.prep.outputs.tag }}
+           cache-from: type=local,src=/tmp/.buildx-cache
+           cache-to: mode=max,type=local,dest=/tmp/.buildx-cache
+
+
+  test_1:
+    name: "Read Notebooks"
+    runs-on: ubuntu-latest
+    needs: ghr_push
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare
+        id: prep
+        uses: ./.github/actions/docker-image-tag
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.REP_PAT }}
+
+      - name: Pull image from DockerHub
+        run: docker pull ${{ steps.prep.outputs.tag }}
+
+      - name: Run image in container
+        run: docker run --rm ${{ steps.prep.outputs.tag }} /bin/bash -c "poetry run nbdev_read_nbs"
+
+  test_2:
+    name: "Check Notebooks Clean"
+    runs-on: ubuntu-latest
+    needs: ghr_push
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare
+        id: prep
+        uses: ./.github/actions/docker-image-tag
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.REP_PAT }}
+
+      - name: Pull image from DockerHub
+        run: docker pull ${{ steps.prep.outputs.tag }}
+
+      - name: Run image in container
```

Comment:
When going through the CI output steps it wasn't clear to me that `Run image in container` step contains the main information.
I would suggest to to use the job name also for the task name (or something very similar).
`Pull image from DockerHub` -> `"Check Notebooks Clean"` The same applies to the other jobs.

Reply:
:boom: